bundles/org.openhab.binding.modbus.studer/src/main/java/org/openhab/binding/modbus/studer/internal/StuderHandler.java

```diff
@@ -0,0 +1,460 @@
+        }
+
+        logger.debug("Setting up regular polling");
+
+        ModbusReadRequestBlueprint request = new ModbusReadRequestBlueprint(slaveAddress,
+                ModbusReadFunctionCode.READ_INPUT_REGISTERS, n, 2, studerConfig.maxTries);
+        long refreshMillis = 3 * 1000;
+        PollTask pollTask = mycomms.registerRegularPoll(request, refreshMillis, 1000, result -> {
+            if (result.getRegisters().isPresent()) {
+                ModbusRegisterArray reg = result.getRegisters().get();
+                handlePolledData(n, reg);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                return;
+            }
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        }, this::handleError);
+        pollTasks.add(pollTask);
+    }
+
+    /**
+     * This method is called each time new data has been polled from the modbus slave
+     * The register array is first parsed, then each of the channels are updated
+     * to the new values
+     *
+     * @param n register readed
+     * @param registers byte array read from the modbus slave
+     */
+    protected void handlePolledData(int n, ModbusRegisterArray registers) {
+        // TODO Auto-generated method stub
+        String hexString = registers.toHexString().toString();
+        Object quantity = parser.hexToFloat(hexString);
+        if (quantity != null) {
+            try {
+                if (type.equals(THING_TYPE_BSP)) {
+                    updateState(CHANNELS_BSP.get(n), new QuantityType<>((float) quantity, UNIT_CHANNELS_BSP.get(n)));
+                } else if (type.equals(THING_TYPE_XTENDER)) {
+                    handlePolledDataXtender(n, quantity);
+                } else if (type.equals(THING_TYPE_VARIOTRACK)) {
+                    handlePolledDataVarioTrack(n, quantity);
+                } else if (type.equals(THING_TYPE_VARIOSTRING)) {
+                    handlePolledDataVarioString(n, quantity);
+                }
+            } catch (NumberFormatException e) {
+                // Do nothing
+            }
+        }
+        resetCommunicationError();
+    }
+
+    /**
+     * This method is called each time new data has been polled from the VarioString slave
+     * The register array is first parsed, then each of the channels are updated
+     * to the new values
+     */
+    @SuppressWarnings("null")
+    protected void handlePolledDataVarioString(int n, Object quantity) {
+        switch (CHANNELS_VARIOSTRING.get(n)) {
+            case CHANNEL_PV_OPERATING_MODE:
+            case CHANNEL_PV1_OPERATING_MODE:
+            case CHANNEL_PV2_OPERATING_MODE:
+                VSMode vsmode = StuderParser.getVSModeByCode((int) (float) quantity);
+                updateState(CHANNELS_VARIOSTRING.get(n),
+                        vsmode == null ? UnDefType.UNDEF : new StringType(vsmode.name()));
```

Comment:
`vsmode` can't be null as `getVSModeByCode()` is declared as NonNull. So, this will always yield false. Same for below.